src/utils/array.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Group an array of T by a field in T.
+ * @param input what to group
+ * @param identityFn what field to group on
```

Comment:
I think the name `identity` or "identity function" may be commonly associated with the identity function, i.e. the one that returns its argument unchanged?

```ts
const identity = <A>(a: A): A => a
```